packages/block-library/src/navigation/block-colors-selector.js

```diff
@@ -58,27 +58,46 @@ const renderToggleComponent = ( { value } ) => ( { onToggle, isOpen } ) => {
 				label={ __( 'Open Colors Selector' ) }
 				onClick={ onToggle }
 				onKeyDown={ openOnArrowDown }
-				icon={ <ColorSelectorIcon color={ value } /> }
+				icon={ <ColorSelectorIcon color={ textColor } backgroundColor={ backgroundColor } /> }
 			/>
 		</ToolbarGroup>
 	);
 };
 
-const renderContent = ( { value, onChange = noop } ) => ( () => {
+const renderContent = ( {
+	textColor,
+	onTextColorChange = noop,
+	backgroundColor,
+	onBackgroundColorChange = noop,
+} ) => ( () => {
 	return (
 		<>
 			<div className="color-palette-controller-container">
 				<ColorPaletteControl
-					value={ value }
-					onChange={ onChange }
+					value={ textColor }
+					onChange={ onTextColorChange }
 					label={ __( 'Text Color' ) }
 				/>
 			</div>
+
+			<div className="color-palette-controller-container">
+				<ColorPaletteControl
+					value={ backgroundColor }
+					onChange={ onBackgroundColorChange }
+					label={ __( 'Background Color' ) }
+				/>
+			</div>
+
+			<ContrastChecker
+				textColor={ textColor }
+				backgroundColor={ backgroundColor }
+				isLargeText={ false }
```

Comment:
In the future will this need to be dynamic to accommodate the font size picker setting?